### ISSUE TITLE
New Anaconda Terms Of Service (TOS)

### ISSUE THREAD

**Issue by 21010** [robocorp/rcc#3]
Anaconda changed its Terms Of Service (TOS), requiring a Business or Enterprise license from organizations with 200 or more employees. It means the usage of repo.anaconda.com and repo.anaconda.cloud requires a commercial license.
Could you please clarify how this affects the usage of RCC?

**Comment by simonmeggle** [comment]
Which channels are you using? 
I understand that community channels like conda-forge are not affected by this. 

https://www.anaconda.com/pricing/terms-of-service-faqs

> Do I have to pay if I use Miniconda?
Because Miniconda is a minimal Anaconda distribution installer that points to the Anaconda repositories by default, the need to pay depends on the package channels you use. If you use the default channel (named “defaults” and points to the Anaconda Repository) and your company has 200 or more employees, you need to purchase a license. 
However, if you use community-maintained package channels on anaconda.org (like **conda-forge** and bioconda) as your preferred channel, you don’t need a license.

> Do commercial fees apply to conda-forge?
No. Our commercial fees do not apply to the user-uploaded packages at [anaconda.org](http://anaconda.org/), which includes conda-forge. We do not build these packages. We host them as a free service to the community – something we’ve done for over 10 years. While commercial fees don’t apply, our Terms of Service do still apply to conda-forge, as they do for any third-party content we host, which is typically the case when an internet service provider hosts third-party content. For example, by using anaconda.org, users agree not to share content that is illegal, harmful, threatening, abusive, harassing, defamatory, obscene, pornographic, offensive, invades another’s privacy, or promotes bigotry, racism, hatred or harm against any individual or group.

```
cat conda.yaml
channels:
  - conda-forge    # <----
...
```

**Comment by 21010** [comment]
Thanks. I use the default rcc setup  which is conda-forge repository.
Is that mean RCC does not connect to anaconda.com at any point?

czw., 5 gru 2024, 14:10 użytkownik Simon Meggle ***@***.***>
napisał:

**Comment by simonmeggle** [comment]
It does. 
But as long as the _community_-maintained channels are used (like **_conda-forge_**, see the FAQ), I do not see a need for a license.

**Comment by 21010** [comment]
I guess rcc connects with Anaconda.org not anaconda.com. Right?

czw., 5 gru 2024, 14:32 użytkownik Simon Meggle ***@***.***>
napisał:

**Comment by simonmeggle** [comment]
Wireshark says so :-) 
![image](https://github.com/user-attachments/assets/db77828f-25b8-4560-becd-a3b3aa587d8a)

**Comment by joshyorko** [comment]
Sorry im late to the party:

Hey @21010 @simonmeggle,

Great discussion here. Let me clarify how RCC handles conda channels and the Anaconda TOS situation.

### TL;DR: You're Fine with conda-forge

**RCC uses `conda.anaconda.org` (NOT `repo.anaconda.com`)** to fetch packages. When your `conda.yaml` specifies `conda-forge` as the channel, you're downloading from the community-maintained conda-forge channel hosted on anaconda.org - which is **free and not subject to the commercial TOS**.

### How It Works

1. **RCC's default endpoint**: `https://conda.anaconda.org/`
2. **Your conda.yaml channel**: `conda-forge`
3. **Resulting URL**: `https://conda.anaconda.org/conda-forge/...`

This is the community-hosted channel, not the commercial "defaults" channel that requires a license.

### The Important Distinction

| Channel | URL | License Required? |
|---------|-----|-------------------|
| `defaults` | `repo.anaconda.com` | Yes (200+ employees) |
| `conda-forge` | `conda.anaconda.org/conda-forge` | **No** |
| `bioconda` | `conda.anaconda.org/bioconda` | **No** |

RCC templates default to `conda-forge`, so you're already using the free community channel.

### Verify Your Setup

Check your `conda.yaml`:

```yaml
channels:
  - conda-forge    # This is FREE - community maintained
  # - defaults     # AVOID this - requires commercial license for 200+ employees
```

RCC will actually warn you if you use `defaults`:

```
Warning: Try to avoid defaults channel, and prefer using conda-forge instead.
```

### If You Want Full Control

In my community fork, you can override the conda endpoint entirely:

**Option 1: Environment Variable**
```bash
export RCC_ENDPOINT_CONDA="https://your-internal-mirror.com/"
```

**Option 2: settings.yaml**
```yaml
endpoints:
  conda: https://your-internal-conda-mirror.com/
```

This lets you point to an internal mirror if your organization requires it.

---

### Why Use My Fork?

My community fork at [github.com/joshyorko/rcc](https://github.com/joshyorko/rcc) includes:

- **Telemetry disabled by default** (no calls to robocorp.com)
- **All endpoints configurable** via env vars or settings.yaml
- **Active maintenance** with community contributions
- **Same conda-forge defaults** as upstream

Download the latest: [github.com/joshyorko/rcc/releases/latest](https://github.com/joshyorko/rcc/releases/latest)

---

### Summary

- Using `conda-forge` channel = **FREE**, no license needed
- RCC fetches from `conda.anaconda.org`, not `repo.anaconda.com`
- Anaconda's commercial TOS only applies to the `defaults` channel
- RCC warns you if you accidentally use `defaults`

You're good to go. Let me know if you have other questions!
